### PR TITLE
proxmark3-iceman: update to v4.16191

### PR DESCRIPTION
**Pull request by linuxgemini**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.3 22E252 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

fyi; trace mode is borked on ventura still so `-vs` is used instead of `-vst`

### CODE REVIEW

File: science/proxmark3-iceman/Portfile [hunk 1]
```diff
@@ -33,14 +33,35 @@ depends_lib-append  port:readline \
...
+# although the upstream repository does not have a ./configure script,
+# the buildsystem itself does rely on things usually found in ./configure
+# like for instance, pkg-config.
+#
+# this is why i (@linuxgemini) has decided to set variables related to
+# configure phase instead of making new ones before passing them
+# to build and destroot envs.
+
+set python_prefix   ${frameworks_dir}/Python.framework/Versions/3.9
+configure.python    ${prefix}/bin/python3.9
```
Comment (reneeotten) [inline science/proxmark3-iceman/Portfile:48]: default Python version in MacPorts is 3.11 and that should be used if possible

File: science/proxmark3-iceman/Portfile [hunk 2]
```diff
@@ -33,14 +33,35 @@ depends_lib-append  port:readline \
...
+
+configure.pkg_config \
+                    ${prefix}/bin/pkg-config
+configure.pkg_config_path-append \
+                    ${python_prefix}/lib/pkgconfig
+
```
Comment (reneeotten) [inline science/proxmark3-iceman/Portfile:54]: this does imply you should have a build dependency on "pkgconfig"?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: science/proxmark3-iceman/Portfile (hunk 2 above)
Comment (linuxgemini) [reply science/proxmark3-iceman/Portfile:54]: Linter warns with `Warning: Dependency port:pkgconfig specified multiple times in depends_build` as the `qt5` PortGroup includes it already:

https://github.com/macports/macports-ports/blob/9842c6d58d2c8fce706598308eb1cfbce0d59b45/_resources/port1.0/group/qt5-1.0.tcl#L39-L40

File: science/proxmark3-iceman/Portfile (hunk 1 above)
Comment (linuxgemini) [reply science/proxmark3-iceman/Portfile:48]: Let me check if upstream has any issues about it, will change then

File: science/proxmark3-iceman/Portfile (hunk 2 above)
Comment (reneeotten) [reply science/proxmark3-iceman/Portfile:54]: right - sorry, I missed that it was already included by the qt5 PortGroup

File: science/proxmark3-iceman/Portfile (hunk 1 above)
Comment (linuxgemini) [reply science/proxmark3-iceman/Portfile]: changed to 3.11

**Pull request merged**